README.md

```diff
@@ -14,12 +14,23 @@ If there is anything you are looking for that is not covered here you should ref
 
 ### File structure ###
 
-You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file.
+You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file. If an enum or protocol is related to the class but has to be known outside define it inside the class file.
```

Comment:
As per our discussion today let's get rid of "If an enum or protocol is related to the class but has to be known outside define it inside the class file." and keep things simple

Maybe add a rationale about "Why are we making enums and protocols into their own file/type"